### PR TITLE
account: Services Refactoring

### PR DESCRIPTION
**Pull request by AcK77**
This PR cleanup our account services implementation.
Since the service use the same "basic" functions in different interfaces, I've made some "Server" class which contains generic calls implementation. That lets us implement more calls without writing more duplicate code.

- IManagerForApplication and IManagerForSystemService are based on the same calls.
- IProfile and IProfileEditor are based on the same calls.
- IAccountServiceForAdministrator, IAccountServiceForApplication and IAccountServiceForSystemService are based on the same calls.

I've cleaned arp:r service and IServiceCreator a bit.
Everything is checked by a quick RE.

### CODE REVIEW

**Review by Thog — APPROVED** [review]
lgtm

File: Ryujinx.HLE/HOS/Services/Account/Acc/ApplicationServiceServer.cs [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        }
+
+        public ResultCode GetLastOpenedUser(ServiceCtx context)
+        {
+            context.Device.System.State.Account.LastOpenedUser.UserId.Write(context.ResponseData);
+
+            return ResultCode.Success;
+        }
+
+        public ResultCode GetProfile(ServiceCtx context, out IProfile iProfile)
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/Services/Account/Acc/ApplicationServiceServer.cs:86]: Why the i prefix on the parameter name?

File: Ryujinx.HLE/HOS/Services/Account/Acc/ApplicationServiceServer.cs [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+            // Doesn't occur in our case.
+            // return ResultCode.NullObject;
+
+            return ResultCode.Success;
+        }
+
+        public ResultCode IsUserRegistrationRequestPermitted(ServiceCtx context)
+        {
+            context.ResponseData.Write(_serviceFlag == AccountServiceFlag.Application ? false : true);
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/Services/Account/Acc/ApplicationServiceServer.cs:114]:
```suggestion
            context.ResponseData.Write(_serviceFlag != AccountServiceFlag.Application);
```

File: Ryujinx.HLE/HOS/Services/Account/Acc/ApplicationServiceServer.cs [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+            return ResultCode.Success;
+        }
+
+        public ResultCode TrySelectUserWithoutInteraction(ServiceCtx context)
+        {
+            if (context.Device.System.State.Account.GetUserCount() != 1)
+            {
+                // Invalid UserId.
+                new UserId(0, 0).Write(context.ResponseData);
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/Services/Account/Acc/ApplicationServiceServer.cs:124]: I suggest adding a `Null` property to `UserId` and then changing this to `UserId.Null.Write(context.ResponseData);`.

**Review by gdkchan — APPROVED** [review]
lgtm, thanks.

**Pull request merged**